### ISSUE TITLE
Fail to install lapack:x64-windows-static

### ISSUE THREAD

**Issue by liupeiqiHN** [microsoft/vcpkg#21512]
**Describe the bug**
Fail to install lapack:x64-windows-static today, my last successful installation was on 2021/10/07, but during this time, my IDE has changed, I upgraded vs2019 to vs2022.

**Environment**
- OS: win10
- Compiler: MSVC 2022
- package: lapack[core]:x64-windows-static -> 3#2
- vcpkg version: cbc22a396 2021-11-17 (13 hours ago)
- vcpkg-tool version: 2021-11-15-ec76affcc987efbf172b7da0b05df711b48fa535

**Failure logs** 
### vcpkg install lapack:x64-windows-static
```
Computing installation plan...
The following packages will be built and installed:
    lapack[core]:x64-windows-static -> 3#2
Detecting compiler hash for triplet x64-windows-static...
Restored 0 packages from C:\Users\LPQ\AppData\Local\vcpkg\archives in 158.5 us. Use --debug to see more details.
Starting package 1/1: lapack:x64-windows-static
Building package lapack[core]:x64-windows-static...
-- Found external ninja('1.10.2').
-- Configuring x64-windows-static
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: ninja -v
    Working Directory: D:/vcpkg/buildtrees/lapack/x64-windows-static-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\lapack\config-x64-windows-static-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:345 (vcpkg_execute_required_process)
  ports/lapack/portfile.cmake:4 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)


Error: Building package lapack:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  package: lapack[core]:x64-windows-static -> 3#2
  vcpkg version: cbc22a396 2021-11-17 (13 hours ago)
  vcpkg-tool version: 2021-11-15-ec76affcc987efbf172b7da0b05df711b48fa535

Additionally, attach any relevant sections from the log files above.


------------------------------
Execution Finished
```

###  config-x64-windows-static-out.log
```
[2/2] cmd /c "cd .. && "D:/vcpkg/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe" "D:/vcpkg/ports/lapack" "-DCMAKE_PREFIX_PATH="D:/vcpkg/packages/lapack_x64-windows-static"" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows-static" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=static" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/vcpkg" "-DZ_VCPKG_ROOT_DIR=D:/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-Ax64" "-G" "Visual Studio 17 2022" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/lapack_x64-windows-static""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "D:/vcpkg/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe" "D:/vcpkg/ports/lapack" "-DCMAKE_PREFIX_PATH="D:/vcpkg/packages/lapack_x64-windows-static"" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x64-windows-static" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=static" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=D:/vcpkg" "-DZ_VCPKG_ROOT_DIR=D:/vcpkg" "-D_VCPKG_INSTALLED_DIR=D:/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-Ax64" "-G" "Visual Studio 17 2022" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/lapack_x64-windows-static""
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
-- The C compiler identification is MSVC 19.30.30705.0
-- The CXX compiler identification is MSVC 19.30.30705.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: D:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: D:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using VCPKG FindLAPACK from package 'lapack-reference'
-- Looking for sgemm_
-- Looking for sgemm_ - not found
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE  
-- Looking for sgemm_
-- Looking for sgemm_ - found
-- Found BLAS: D:/vcpkg/installed/x64-windows-static/lib/openblas.lib  
-- Looking for cheev_
-- Looking for cheev_ - not found
-- Looking for cheev_
-- Looking for cheev_ - not found
CMake Error at D:/vcpkg/installed/x64-windows-static/share/lapack/FindLAPACK.cmake:522 (message):
  A required library with LAPACK API not found.  Please specify library
  location.
Call Stack (most recent call first):
  D:/vcpkg/installed/x64-windows-static/share/lapack/vcpkg-cmake-wrapper.cmake:9 (_find_package)
  D:/vcpkg/scripts/buildsystems/vcpkg.cmake:742 (include)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
See also "D:/vcpkg/buildtrees/lapack/x64-windows-static-rel/CMakeFiles/CMakeOutput.log".
See also "D:/vcpkg/buildtrees/lapack/x64-windows-static-rel/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.
```

**Comment by liupeiqiHN** [comment]
I failed to install lapack:x64-windows too:
```
Computing installation plan...
The following packages will be built and installed:
  * blas[core]:x64-windows -> 1#1
    lapack[core]:x64-windows -> 3#2
  * lapack-reference[blas-select,core,noblas]:x64-windows -> 3.8.0#6
  * vcpkg-gfortran[core]:x64-windows -> 3#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\LPQ\AppData\Local\vcpkg\archives in 415 us. Use --debug to see more details.
Starting package 1/4: blas:x64-windows
Building package blas[core]:x64-windows...
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\LPQ\AppData\Local\vcpkg\archives\3c\3c90430abfb6ba050bbd3c90fcfe0b9d0932165a6e47e40ab14794c0c4bf8bd4.zip
Installing package blas[core]:x64-windows...
Elapsed time for package blas:x64-windows: 5.354 s
Starting package 2/4: vcpkg-gfortran:x64-windows
Building package vcpkg-gfortran[core]:x64-windows...
-- The Fortran compiler identification is unknown
-- No Fortran compiler found on the PATH. Using MinGW gfortran!
-- Using cached msys-mingw-w64-x86_64-gcc-fortran-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gcc-libgfortran-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gcc-libs-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gmp-6.2.0-1-any.pkg.tar.xz.
-- Using cached msys-mingw-w64-x86_64-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-winpthreads-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-mpc-1.1.0-1-any.pkg.tar.xz.
-- Using cached msys-mingw-w64-x86_64-mpfr-4.1.0-2-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gcc-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-binutils-2.34-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-crt-git-8.0.0.5966.f5da805f-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-headers-git-8.0.0.5966.f5da805f-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-libiconv-1.16-2-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-windows-default-manifest-6.4-3-any.pkg.tar.xz.
-- Using cached msys-mingw-w64-x86_64-zlib-1.2.11-8-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-zstd-1.4.5-1-any.pkg.tar.zst.
-- Using cached msys-gzip-1.10-1-x86_64.pkg.tar.xz.
-- Using cached msys-bash-4.4.023-2-x86_64.pkg.tar.xz.
-- Using cached msys-diffutils-3.7-1-x86_64.pkg.tar.xz.
-- Using cached msys-coreutils-8.32-1-x86_64.pkg.tar.xz.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-1-x86_64.pkg.tar.xz.
-- Using cached msys-libpcre-8.44-1-x86_64.pkg.tar.xz.
-- Using cached msys-gawk-5.1.0-1-x86_64.pkg.tar.xz.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.0-1-x86_64.pkg.tar.xz.
-- Using cached msys-libreadline-8.0.004-1-x86_64.pkg.tar.xz.
-- Using cached msys-ncurses-6.2-1-x86_64.pkg.tar.xz.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-1-x86_64.pkg.tar.xz.
-- Using cached msys-libintl-0.19.8.1-1-x86_64.pkg.tar.xz.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-9.3.0-1-x86_64.pkg.tar.xz.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using msys root at D:/vcpkg/downloads/tools/msys2/456f7716e2893cfb
-- Installing: D:/vcpkg/packages/vcpkg-gfortran_x64-windows/share/vcpkg-gfortran/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\LPQ\AppData\Local\vcpkg\archives\e0\e0d02c292c6054ebd6b45fada9ab16c9eff744cbc5c562c73cf0779126d0bf60.zip
Installing package vcpkg-gfortran[core]:x64-windows...
Elapsed time for package vcpkg-gfortran:x64-windows: 1.465 s
Starting package 3/4: lapack-reference:x64-windows
Building package lapack-reference[blas-select,core,noblas]:x64-windows...
-- Using cached Reference-LAPACK-lapack-v3.8.0.tar.gz.
-- Cleaning sources at D:/vcpkg/buildtrees/lapack-reference/src/v3.8.0-954f10683f.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/vcpkg/downloads/Reference-LAPACK-lapack-v3.8.0.tar.gz
-- Using source at D:/vcpkg/buildtrees/lapack-reference/src/v3.8.0-954f10683f.clean
-- The Fortran compiler identification is unknown
-- No Fortran compiler found on the PATH. Using MinGW gfortran!
-- Using cached msys-mingw-w64-x86_64-gcc-fortran-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gcc-libgfortran-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gcc-libs-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gmp-6.2.0-1-any.pkg.tar.xz.
-- Using cached msys-mingw-w64-x86_64-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-winpthreads-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-mpc-1.1.0-1-any.pkg.tar.xz.
-- Using cached msys-mingw-w64-x86_64-mpfr-4.1.0-2-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-gcc-10.2.0-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-binutils-2.34-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-crt-git-8.0.0.5966.f5da805f-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-headers-git-8.0.0.5966.f5da805f-1-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-libiconv-1.16-2-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-windows-default-manifest-6.4-3-any.pkg.tar.xz.
-- Using cached msys-mingw-w64-x86_64-zlib-1.2.11-8-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-x86_64-zstd-1.4.5-1-any.pkg.tar.zst.
-- Using cached msys-gzip-1.10-1-x86_64.pkg.tar.xz.
-- Using cached msys-bash-4.4.023-2-x86_64.pkg.tar.xz.
-- Using cached msys-diffutils-3.7-1-x86_64.pkg.tar.xz.
-- Using cached msys-coreutils-8.32-1-x86_64.pkg.tar.xz.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-1-x86_64.pkg.tar.xz.
-- Using cached msys-libpcre-8.44-1-x86_64.pkg.tar.xz.
-- Using cached msys-gawk-5.1.0-1-x86_64.pkg.tar.xz.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.0-1-x86_64.pkg.tar.xz.
-- Using cached msys-libreadline-8.0.004-1-x86_64.pkg.tar.xz.
-- Using cached msys-ncurses-6.2-1-x86_64.pkg.tar.xz.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-1-x86_64.pkg.tar.xz.
-- Using cached msys-libintl-0.19.8.1-1-x86_64.pkg.tar.xz.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-9.3.0-1-x86_64.pkg.tar.xz.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using msys root at D:/vcpkg/downloads/tools/msys2/456f7716e2893cfb
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Fixing pkgconfig file: D:/vcpkg/packages/lapack-reference_x64-windows/lib/pkgconfig/blas.pc
-- Fixing pkgconfig file: D:/vcpkg/packages/lapack-reference_x64-windows/lib/pkgconfig/lapack.pc
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-2-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst.
-- Using msys root at D:/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Fixing pkgconfig file: D:/vcpkg/packages/lapack-reference_x64-windows/debug/lib/pkgconfig/blas.pc
-- Fixing pkgconfig file: D:/vcpkg/packages/lapack-reference_x64-windows/debug/lib/pkgconfig/lapack.pc
-- Installing: D:/vcpkg/packages/lapack-reference_x64-windows/share/lapack-reference/copyright
-- Performing post-build validation
Import libs were not present in D:\vcpkg\packages\lapack-reference_x64-windows\debug\lib
If this is intended, add the following line in the portfile:
    SET(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
Import libs were not present in D:\vcpkg\packages\lapack-reference_x64-windows\lib
If this is intended, add the following line in the portfile:
    SET(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
Found 2 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile:
    D:\vcpkg\ports\lapack-reference\portfile.cmake
-- Performing post-build validation done
Stored binary cache: C:\Users\LPQ\AppData\Local\vcpkg\archives\89\897a902f692f23b21d92a530d62bba1a9b732b4ef3764cc264031d6f7461c238.zip
Installing package lapack-reference[blas-select,core,noblas]:x64-windows...
Elapsed time for package lapack-reference:x64-windows: 1.726 min
Starting package 4/4: lapack:x64-windows
Building package lapack[core]:x64-windows...
-- Found external ninja('1.10.2').
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: ninja -v
    Working Directory: D:/vcpkg/buildtrees/lapack/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\lapack\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:345 (vcpkg_execute_required_process)
  ports/lapack/portfile.cmake:4 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)


Error: Building package lapack:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  package: lapack[core]:x64-windows -> 3#2
  vcpkg version: cbc22a396 2021-11-17 (13 hours ago)
  vcpkg-tool version: 2021-11-15-ec76affcc987efbf172b7da0b05df711b48fa535

Additionally, attach any relevant sections from the log files above.


------------------------------
Execution Finished
```

**Comment by dg0yt** [comment]
Maybe same as #20323 ?

**Comment by LilyWangLL** [comment]
Duplicate with issue #20323, this error will be fixed in PR #19608. I will close this issue, please pay attention to the progress of issue #20323.
